dylint/src/library_packages/mod.rs

```diff
@@ -222,15 +215,15 @@ fn library_packages_from_dylint_metadata(
 }
 
 pub fn from_dylint_toml(opts: &opts::Dylint) -> Result<Vec<Package>> {
-    if_chain! {
-        if let Some(metadata) = cargo_metadata(opts)?;
+    if let Some(metadata) = cargo_metadata(opts)? {
         let _ = config::try_init_with_metadata(metadata)?;
-        if let Some(table) = config::get();
-        then {
+        if let Some(table) = config::get() {
             library_packages_from_dylint_toml(opts, metadata, table)
         } else {
             Ok(vec![])
         }
+    } else {
+        Ok(vec![])
```

Comment:
This translation is worth another look. I'm pretty sure it is correct, but the `let _` assignment in between the two `if let` statements is interesting. I think this does preserve the original code behavior though.